File: 3206-alternating-groups-i/3206-alternating-groups-i.java
class Solution {
    public int numberOfAlternatingGroups(int[] colors) {
        int count = 0;
        int n = colors.length;
        for (int i = 0; i < n; i++) {
            if (colors[i] != colors[(i-1 + n) % n] && colors[i] != colors[(i+1) % n]) count++;
        }
        return count;
    }
}

File: 3206-alternating-groups-i/3206-alternating-groups-i.cpp
class Solution {
public:
    int numberOfAlternatingGroups(vector<int>& colors) {
        int count = 0;
        colors.push_back(colors[0]);
        colors.push_back(colors[1]);
        int n = colors.size();

        for(int i=2;i<n;i++)
        {
            if(colors[i] != colors[i-1] && colors[i-1] != colors[i-2])
            {
                count++;
            }
        }
        return count;
    }
};